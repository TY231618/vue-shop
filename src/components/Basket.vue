<template>
  <div class="basket">
    <h4>Basket:</h4>
    <slot v-if="basket.length > 0">
      <div class="basket-item" v-for="product in basket">
        <div class="item-name">
          {{product.name}}
        </div>
        <div>
          <button class="waves-effect waves-light btn" @click="removeItemFromBasket(product)" ><i class="material-icons">cancel</i></button>
          <button class="waves-effect waves-light btn"><router-link to="/checkout">Checkout</router-link></button>
        </div>
      </div>
      <p class="basket-size">number of Items: {{basket.length}}</p>
      <p>Total: Â£ {{total}}</p>
      <p>shipping cost: {{shipping}}</p>
    </slot>
    <slot v-if="basket.length == 0">
      <p>Buy something tight ass</p>
    </slot>
  </div>
</template>

<script>
  import { mapState } from 'vuex'

  export default {

    name: 'basket',
    props: [
      'shipping',
      'product'
    ],
    methods: {
      removeItemFromBasket(product) {
        this.$store.commit('removeItemFromBasket', product)
      }
    },
    computed: {
      ...mapState([
        'basket',
        'total'
      ])
    }
  }
</script>

<style>

</style>
