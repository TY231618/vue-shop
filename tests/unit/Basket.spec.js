import { mount, shallow } from '@vue/test-utils';
import Basket from '../../src/components/Basket';

describe('Basket', () => {

  const basket = [{name: 'dog shirt'}, {name: 'cat shirt'}];

  const component = shallow(Basket, {
    propsData: {
      basket: basket
    }
  });

  it('should render successfully', () => {

    expect(component.exists()).toEqual(true);
  })

  it('should match snapshot', () => {
    // component.setProps({basket:[{name: 'dog shirt'}]})
    expect(component.html()).toMatchSnapshot();
    expect(component.find('.basket-size').text()).toEqual('number of Items: 2')
  })
})