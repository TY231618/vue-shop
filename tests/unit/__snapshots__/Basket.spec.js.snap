// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket should match snapshot 1`] = `
<div class="basket">
    <h4>Basket:</h4>
    <div class="basket-item">
        <div class="item-name">
            dog shirt
        </div>
        <div>
            <button class="waves-effect waves-light btn">
                <i class="material-icons">cancel</i>
            </button>
        </div>
    </div>
    <div class="basket-item">
        <div class="item-name">
            cat shirt
        </div>
        <div>
            <button class="waves-effect waves-light btn">
                <i class="material-icons">cancel</i>
            </button>
        </div>
    </div>
    <p class="basket-size">number of Items: 2</p>
    <p>Total: Â£ </p>
    <p>shipping cost: </p>
    <!---->
</div>
`;
